### PR TITLE
ISO images by default are uploaded as Raw/Thin Provision. And if you …

### PR DESCRIPTION
**Pull request by sermakov-orion**
ISO images by default are uploaded as Raw/Thin Provision. And if you later move such disk to a block storage (FC/iSCSI) then it will be converted to Cow/Thin Provision. And this might corrupt the disk. To avoid this issue we suggested to always upload ISO images as Raw/Preallocated. In this case no conversion needed while moving to a block storage. And disk works fine in both types (block and non-block) storages.

## Are you the owner of the code you are sending in, or do you have permission of the owner?

y

### CODE REVIEW

**Review by sandrobonazzola — CHANGES_REQUESTED** [review]
This branch has conflicts that must be resolved

**Comment by sermakov-orion** [comment]
@sandrobonazzola 
The conflicts were resolved. The branch is up to date now

**Review by sandrobonazzola — APPROVED** [review]

File: frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/models/storage/UploadImageModel.java [hunk 1]
```diff
@@ -371,12 +372,18 @@ public boolean flush() {
...
-                    VolumeFormat.COW : getImageInfoModel().getFormat());
-            diskImage.setVolumeType(AsyncDataProvider.getInstance().getVolumeType(
-                    diskImage.getVolumeFormat(),
-                    getDiskModel().getStorageDomain().getSelectedItem().getStorageType(), null, null));
-            diskImage.setContentType(getImageInfoModel().getContentType());
+            VolumeFormat volumeFormat = diskModel.getIsIncrementalBackup().getEntity() ?
+                    VolumeFormat.COW : getImageInfoModel().getFormat();
+            diskImage.setVolumeFormat(volumeFormat);
+            DiskContentType contentType = getImageInfoModel().getContentType();
+            if (volumeFormat != VolumeFormat.COW && contentType == DiskContentType.ISO) {
```
Comment (dupondje) [inline frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/models/storage/UploadImageModel.java:379]: I think we can just add the additional if:
```
if (diskImage.getVolumeFormat() != VolumeFormat.COW && getImageInfoModel().getContentType() == DiskContentType.ISO) {
                diskImage.setVolumeType(VolumeType.Preallocated);
            }
```
And skip the other changes :)

**Review by dupondje — CHANGES_REQUESTED** [review]
Can be rewritten more shortly.

File: frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/models/storage/UploadImageModel.java (hunk 1 above)
Comment (sermakov-orion) [reply frontend/webadmin/modules/uicommonweb/src/main/java/org/ovirt/engine/ui/uicommonweb/models/storage/UploadImageModel.java]: @dupondje Thanks for the suggestion.
I applied the changes (with additional comments).
Could you please review once more?
Thanks in advance.

**Review by dupondje — APPROVED** [review]

**Pull request merged**